files/common/var/lib/delphix-platform/os-migration

```diff
@@ -129,6 +129,15 @@ function perform_migration() {
 	__trigger_unset_stress_option \
 		"STRESS_MIGRATION_FAIL_AFTER_DOMAIN0_IMPORT"
 
+	#
+	# The ashift value was stored as a user property on the domain0
+	# dataset so we can't read it until after we've imported the pool.
+	# A value of 0 is the default setting.
+	#
+	ashift=$(zfs get -Ho value dlpx:ashift domain0)
+	zpool set ashift=$ashift domain0 ||
+		die "Failed to set ashift value on domain0"
```

Comment:
- Could we skip setting this if ashift is 0 or empty?
- Do we need to re-import domain0 for this setting to take effect?

Reply:
We could skip the setting if it's 0 but is there a reason we would want to add more code here? We don't need to re-import for this to take effect.

Reply:
We don't need to do this check per-say, but the idea was to keep changes between migrated and freshly created systems to a minimum.

In the same spirit, we might also want to unset `dlpx:ashift` after we are done with this. 

I'm considering creating a document outlining all of the customizations we do to customer engines, so we can keep track of them in one place. This will make it easier to make sure we do the right testing when needed.

Reply:
I think it also might be worth adding a case handling a case where `dlpx:ashift` doesn't get created. What I'm imagining is: `ashift` is non-zero, but `migrate_config.py` fails to create `dlpx:ashift`, and then in this script `$(zfs get -Ho value dlpx:ashift domain0)` fails silently and `ashift` is gets set to 0 because of bash default values. Then we'd inadvertently change `ashift` to 0.